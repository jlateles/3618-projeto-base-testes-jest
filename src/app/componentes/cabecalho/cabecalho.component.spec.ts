import { CabecalhoComponent } from './cabecalho.component';
import {ComponentFixture, TestBed } from '@angular/core/testing';

describe('Componente de cabeçalho', () => {
  let component: CabecalhoComponent;
  let fixture: ComponentFixture<CabecalhoComponent>

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [CabecalhoComponent],
    })
    fixture = TestBed.createComponent(CabecalhoComponent);
    component = fixture.componentInstance;
  });

  it('Deve estar criado', () => {
    expect(component).toBeTruthy();
  })

  it('Deve definir as propriedades alt e src', () => {
    expect(component.alt).toBeDefined();
    expect(component.src).toBeDefined();
  })

  it('Deve renderizar o conteúdo baseado nas propriedades src e alt', () => {
    component.src = 'https://example.com/teste-image.jpg';
    component.alt = 'Imagem teste';
    expect(component).toMatchSnapshot();
  })

})
