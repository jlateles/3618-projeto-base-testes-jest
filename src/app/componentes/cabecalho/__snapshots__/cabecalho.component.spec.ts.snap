// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de cabeçalho Deve renderizar o conteúdo baseado nas propriedades src e alt 1`] = `
CabecalhoComponent {
  "__ngContext__": 4,
  "alt": "Imagem teste",
  "src": "https://example.com/teste-image.jpg",
}
`;
